core/src/main/java/org/elasticsearch/threadpool/ThreadPoolStats.java

```diff
@@ -188,6 +206,7 @@ public void writeTo(StreamOutput out) throws IOException {
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
         builder.startObject(Fields.THREAD_POOL);
+        Collections.sort(stats);
```

Comment:
fair enough, I don’t think toXContent should change state (Collection.sort  changes the lists it gets). The class become thread unsafe (while it used to be thread safe). You can also sort in ThreadPool#stats, if you prefer?

> On 31 Aug 2015, at 13:14, Tanguy Leroux <notifications@github.com> wrote:
> 
> In core/src/main/java/org/elasticsearch/threadpool/ThreadPoolStats.java:
> 
> > @@ -188,6 +206,7 @@ public void writeTo(StreamOutput out) throws IOException {
> >      @Override
> >      public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
> >          builder.startObject(Fields.THREAD_POOL);
> > +        Collections.sort(stats);
> 
> I added it here because I just want the XContent output to be sorted. I think we don't really care otherwise and thus don't need to sort everytime we instantiate a ThreadPoolStats object. But I can move it if you really prefer.
> 
> —
> Reply to this email directly or view it on GitHub.
>